modules/search/class-jetpack-instant-search.php

```diff
@@ -0,0 +1,367 @@
+<?php
+/**
+ * Jetpack Search: Instant Front-End Search and Filtering
+ *
+ * @since 8.3.0
+ * @package jetpack
+ */
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Constants;
+
+class Jetpack_Instant_Search extends Jetpack_Search {
+
+	/**
+	 * Loads the php for this version of search
+	 *
+	 * @since 8.3.0
+	 */
+	public function load_php() {
+		require_once dirname( __FILE__ ) . '/class.jetpack-search-template-tags.php';
+		require_once JETPACK__PLUGIN_DIR . 'modules/widgets/search.php';
+
+		if ( class_exists( 'WP_Customize_Manager' ) ) {
+			require_once dirname( __FILE__ ) . '/class-jetpack-search-customize.php';
+			new Jetpack_Search_Customize();
+		}
+	}
+
+	/**
+	 * Setup the various hooks needed for the plugin to take over search duties.
+	 *
+	 * @since 5.0.0
+	 */
+	public function init_hooks() {
+		if ( ! is_admin() ) {
+			add_filter( 'posts_pre_query', array( $this, 'filter__posts_pre_query' ), 10, 2 );
+			add_action( 'parse_query', array( $this, 'action__parse_query' ), 10, 1 );
+
+			add_action( 'init', array( $this, 'set_filters_from_widgets' ) );
+
+			add_action( 'wp_enqueue_scripts', array( $this, 'load_assets' ) );
+			add_action( 'wp_footer', array( $this, 'print_instant_search_sidebar' ) );
+		} else {
+			add_action( 'update_option', array( $this, 'track_widget_updates' ), 10, 3 );
+		}
+
+		add_action( 'widgets_init', array( $this, 'register_jetpack_instant_sidebar' ) );
+		add_action( 'jetpack_deactivate_module_search', array( $this, 'move_search_widgets_to_inactive' ) );
+	}
+
+	/**
+	 * Loads assets for Jetpack Instant Search Prototype featuring Search As You Type experience.
+	 */
+	public function load_assets() {
+		$script_relative_path = '_inc/build/instant-search/jp-search.bundle.js';
```

Comment:
Do we want the ability to load the non minified version for debug purposes? Might be worth using get_file_url_for_environment function to determine if the min version should be used. I note this is bundled so may not be applicable :)

https://github.com/Automattic/jetpack/blob/5faa321993c4ca955df871f5a77cd89fa411bca5/packages/assets/src/class-assets.php#L36

Reply:
Note this is non-blocker so going to resolve it, if desired it will be a new PR/task